django/core/management/commands/makemessages.py

```diff
@@ -363,11 +374,16 @@ def handle(self, *args, **options):
         # Build locale list
         looks_like_locale = re.compile(r'[a-z]{2}')
         locale_dirs = filter(os.path.isdir, glob.glob('%s/*' % self.default_locale_path))
+
+        # Here all_locales will already be normalized
```

Comment:
This is just not true - there is no "normalization" happening here. If anything it is an incredibly lax check that the locale folder name starts with two lowercase characters. This makes no guarantees.